lucet-module-data/src/module_data.rs

```diff
@@ -52,6 +61,24 @@ impl<'a> ModuleData<'a> {
         &self.globals_spec
     }
 
+    // Function index here is a different index space than `get_func_from_idx`, which
+    // uses function index as an index into a table of function elements.
+    //
+    // This is an index of all functions in the module.
+    pub fn get_signature(&self, fn_id: u32) -> Option<&Signature> {
+        self.function_info.get(fn_id as usize).and_then(|func| {
+            self.signatures().get(func.signature.as_u32() as usize)
```

Comment:
if the function exists the signature will as well, right - otherwise its a malformed module data? i'd add a `expect` on the signatures().get() so that if this happens we detect it by crashing in tests, hopefully

Reply:
:+1: good call